### ISSUE TITLE
[Bug]: Unclear error message from `check_col_not_nan` check

### ISSUE THREAD

**Issue by TomDonoghue** [NeurodataWithoutBorders/nwbinspector#241]
### What happened?

I ran NWBinspector on some files, and got an unclear error message from the `check_col_not_nan` check, which was raised as an error. From some detective work, we realized this is because in an Electrode Table I do have a column of all NaN (as a placeholder, I had filled the impendences with NaN). 

NWBinspector is right to flag this, but it seems to have trouble reporting this in a clear way, and prints out this long error stack, from which it is tricky to figure out which table / column raised the error. If possible, it would be nice if this error could be caught in a way to be more clear about what the issue is:

<img width="1183" alt="Screen Shot 2022-07-26 at 6 00 46 PM" src="https://user-images.githubusercontent.com/7727566/181119967-2cc3c1f1-4dd9-443f-8204-e970d12e1dd6.png">

### Steps to Reproduce

```shell
This happened from an NWB file with a column of NaNs. If you need a test file, I can make and add one.
```

### Operating System

macOS

### Python Executable

Conda

### Python Version

3.8

### Usage

Command Line Interface

### Were you streaming with ROS3?

No

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this bug was not already [reported](https://github.com/neurodatawithoutborders/nwbinspector/issues)?
- [X] To the best of your ability, have you ensured this is a bug within the code that checks the NWBFile, rather than a bug in the NWBFile reader (e.g., [PyNWB](https://github.com/NeurodataWithoutBorders/pynwb) or [MatNWB](https://github.com/NeurodataWithoutBorders/matnwb))?

**Comment by CodyCBakerPhD** [comment]
@TomDonoghue Thanks for the report!

Sorry you ran into this issue. This is a very recently added check so you're probably one of the first to try running the inspector with the most recent release that includes it.

As far as why the report appears that way (in particular, non very descriptive), we like to capture runtime errors during the check functions and include them in the same report along with any of the non-failing checks so there's no delay in providing at least _some_ feedback about the files. To that extent, we record the full traceback (much like a logger would) of the error so we can do our best to make a fix quickly once the issue is raised. That error stack is mostly for the developer trying to fix it and not intended to be the burden of the user to try to fix - I'll think on attaching some messages or other secondary information to make that clearer to the user, though, thanks for pointing that out!

On that note I've tried a first fix #242, could you try to checkout out that branch and rerun the inspector to see if it's fixed? (`git clone https://github.com/neurodatawithoutborders/nwbinspector` then `git checkout fix_table_col_nan_check` then `pip install -e nwbinspector` or just `pip install git+https://github.com/neurodatawithoutborders/nwbinspector@fix_table_col_nan_check`)

**Comment by TomDonoghue** [comment]
Running my file with the `fix_table_col_nan_check` branch seems to fix things to run as expected!

I no longer get an error log, and I do see what I assume to be a correct report of the issue:

<img width="683" alt="Screen Shot 2022-07-27 at 10 01 51 AM" src="https://user-images.githubusercontent.com/7727566/181266954-efde9582-3114-4086-895d-bbb0c580313d.png">

Let me know if there is anything else I should check!

**Comment by TomDonoghue** [comment]
For the issue of the inspector printing out runtime error logs, does it make sense to have something like a `debug` flag, which if true, prints the full error as current, but if false squashes the error and just says something like: "An error was encountered, set debug=True to see details"?

**Comment by CodyCBakerPhD** [comment]
Great, thanks for checking! 

This particular error also has a secondary influence - the 'imp' column was recently altered to be optional in NWB Schema 2.5.0, since it being previously required made it a _very_ common practice to just set all columns like that to NaN. If using latest PyNWB you therefore shouldn't need to even specify that column at all (so this check message is safe for that version) but it's also a good point that I should probably tell the check to skip whenever the user is writing the file using older schema versions... 

Just for full context of this Issue, what version of PyNWB did you use to write the file?

**Comment by CodyCBakerPhD** [comment]
> For the issue of the inspector printing out runtime error logs, does it make sense to have something like a debug flag, which if true, prints the full error as current, but if false squashes the error and just says something like: "An error was encountered, set debug=True to see details"?

That could be something to consider, definitely! I'll splinter that into a separate isssue.

**Comment by TomDonoghue** [comment]
For the `imp` column in particular, I chatted briefly on this with Ben, and he mentioned this is no longer required. The files I was testing were converted a little while ago, and where _at most_ pynwb 2.0, but I think they were actually before that. From memory, when I set up the file conversion, this column was required (and I wasn't properly adding the electrode metadata anyways, so I dummy coded it). I should be able to update the conversion to simply not include the impedence column.
